Object coordinates R2: rebased from master in 2018 Re #2559, #6078

josephsl opened the pull request:
Hi,
As requested by @MichaelDCurran in #6078:

### Link to issue number:
Fixes #2559

### Summary of the issue:
Play object coordiantes R2: rebased on 2018 master (see #2559 for details).

### Description of how this pull request fixes the issue:
See #6078 discussion history.

### Testing performed:
Same as #6078.

### Known issues with pull request:
None, although @LeonardDer raised concerns which should be addressed.

### Change log entry:
Same as #6078, although the scope can cover focus movement as well.

josephsl committed:
Object coordinates: It is now possible to hear object coordinates when using object navigation commands (rebased with master in June 2016). re #2559

josephsl committed:
NVDAObjects: obj coordinate announcement will be handled via becomeNavObj event. re #2559

As object coordinate announcement will be done regardless of focus movement, it was decided to let becomeNavObject event call call object coordinate player function in screen explorer. This means global commands will not use this method (this decisoin was made after talking to Jamie Teh and others).

josephsl committed:
Object coordinates: obj coordinates checkbox now follows guiHelper services. re #2559

josephsl committed:
Object coordinates: updated copyright headers, removed unused imports.

josephsl committed:
Config: play obj coordinates validator now lives in the new config spec module (after rebasing with master). re #2559.

josephsl committed:
Object coordinates: applied multi-monitor fix from NVDA 2017.1. re #2559.

In NVDA 2017.1, it is possible to hear mouse coordinates on multi-monitor systems. Thus object coordinate announcement function has been modified to take advantage of this. To ensure backward compatibility, minimum screen position paremeter will be a keyword argument, ready to become mandatory if obj coordinate announcement from multi-monitor set ups is desired.

josephsl committed:
Object coordinates: do not play the tone if the object ia offscreen. re #2559

josephsl committed:
User guide: add anchors for object coordinate checkbox. re #2559

josephsl committed:
Update copyright years

josephsl committed:
Object coordinates: do not detect brightness, copyright year update. Re #2559.

Reviewed by Mick Curran (NV Access)L due to some issues, object brightness won't make sense when playing coordinates, so tell coordinates player function to ignore this.
Also updated copyright year.

josephsl committed:
Mouse handler: update copyright years

josephsl committed:
Mouse handler: set copyright header to that of NVDA 2018.2 (to be changed in 2018.3)

josephsl committed:
Mouse handler: restore removed imports.

josephsl committed:
User guide: clarify the scope of object coordinates checkbox. Re #2559.

josephsl committed:
Screen explorer/object coordinates: return early if no location is provided. Re #2559.

In some cases, obj.location is None, which means TypeError is raised because there's nothing to iterate ovdr. Thus, for cases like this, do not play coordinate beep.

josephsl committed:
Object coordinates: play coordinate tone first. Re #2559.

Reviewed by @LeonardDer: if braille is set to auto-tether and if this is a focused object, tones will not play because the event returns early. To get around this, play the tone first.

josephsl commented:
Hi,

In addition to rebasing on master, it also removes mouseHandler.playAudioCoordinates function as no other module uses this function in Core. Mouse handler itself will call screen explorer function.

Thanks.

leonardder commented on source/screenExplorer.py:
```diff
@@ -1,8 +1,68 @@
+	@type obj: int
+	"""
+	# Have a fake wx.Point in case min pos is not defined on multiple monitors.
+	if screenMinPos is None:
```
Why should this be allowed? See my comment below about the way you calculate the screen width and height.

josephsl replied:
Hi,

This came about due to wx.Point fix in order to support mouse tracking on multiple monitors (Reef knows more about this than I do).

Thanks.

leonardder replied:
May be we should step away from getting this information from wx and use [GetSystemMetrics](https://msdn.microsoft.com/en-us/library/windows/desktop/ms724385(v=vs.85).aspx) instead?

leonardder replied:
To clarify, see the SM_CXVIRTUALSCREEN, SM_CYVIRTUALSCREEN, SM_XVIRTUALSCREEN and SM_YVIRTUALSCREEN. If I understand the code correctly (and probably I don't), throwing these four constants at GetSystemMetrics would make mouseHandler.getTotalWidthAndHeightAndMinimumPosition obsolete.

leonardder commented on source/screenExplorer.py:
```diff
@@ -1,8 +1,68 @@
+	if screenMinPos is None:
+		screenMinPos = wx.Point()
+	# Do not play any tone if offscreen.
+	if 0 <= x <= screenWidth or 0 <= y <= screenHeight:
```
It is a matter of style, but personally I'd prefer this to return early.

josephsl replied:
Hi,

Ah, the chained comparison: my message in that fragment was that it should play coordinate beep as long as x and y are referring to things on screen. It was done like this because in browse mode, things will show up offscreen.

Thanks.

feerrenrut replied:
Yeah, I would also prefer that the logic in this `if` statement was reversed and just had an early return. This reduces the indentation level of everything below.

leonardder commented on source/screenExplorer.py:
```diff
@@ -1,8 +1,68 @@
+			startY=min(max(y-blurFactor,0),screenHeight)+screenMinPos.y
+			width=min(blurFactor+1,screenWidth)
+			height=min(blurFactor+1,screenHeight)
+			grey=screenBitmap.rgbPixelBrightness(mouseHandler.scrBmpObj.captureImage( startX, startY, width, height)[0][0])
```
Could you move the scrBmpObj to screenExplorer as well? It currently makes no sense within the scope of mouseHandler.

josephsl replied:
Hi,

Hmmm, the object itself is initialized and nullified when mouse handler starts and ends, respectively. We need to gather more feedback regarding this, as it could break features if this is moved without careful thinking. Thanks.

feerrenrut replied:
There are no other references to `scrBmpObj` outside of `mouseHandler.py`, you can confirm with `git grep -i scrBmpObj`.

Are you worried that addons are using it? My preference would be to understand its usage, and what the intention was behind initialising it this way, with the goal of moving it out of `mouseHandler`

leonardder commented on source/screenExplorer.py:
```diff
@@ -1,8 +1,68 @@
+	l,t,w,h=obj.location
+	x = l+(w/2)
+	y = t+(h/2)
+	screenWidth, screenHeight = api.getDesktopObject().location[-2:]
```
This was actually the main concern I mentioned earlier. Could you please bring this in line with how mouseHandler does this (i.e. use mouseHandler.getTotalWidthAndHeightAndMinimumPosition)

I think it makes sense to move some functionality to a new module. May be locationHelper could be a fit for this. I propose at least moving the following functions, as they aren't specific to the mouseHandler.

* getTotalWidthAndHeightAndMinimumPosition
* getMinMaxPoints

josephsl replied:
Hi,

Code reorg: I propose putting this as part of Python 3 transition.

Thanks.

leonardder commented on source/screenExplorer.py:
```diff
@@ -1,8 +1,68 @@
+	screenWidth, screenHeight = api.getDesktopObject().location[-2:]
+	try:
+		playLocationCoordinates(x, y, screenWidth, screenHeight, detectBrightness=False)
+	except AttributeError:
```
What could cause the AttributeError here?

josephsl replied:
Hi,

Bad assumptions from my part: it was added because it was originally part of mouse handler. Testing shows this is no needed anymore.

Thanks for catching this and helping me update my assumptions.

josephsl commented on source/NVDAObjects/__init__.py:
```diff
@@ -974,6 +975,9 @@ def event_becomeNavigatorObject(self, isFocus=False):
 		@type isFocus: bool
 		"""
+		# #2559: coordinate tone playback takes priority.
+		if config.conf["presentation"]["playObjectCoordinates"]:
```
Hi,

event_becomeNavigatorObject was chosen as it can cover a whole bunch of scenarios. The original idea was to modify global commands, but I and others realized that this will not scalewell, hence this method.

Thanks.

josephsl committed:
Object coordinates/Screen Explorer and GUI: address review actions.

Reviewed by @LeonardDer (Babbage) and catching log errors:
* screenExplorer.playObjectCoordinates: catching attribute error is no longer necessary (born from old assumptions as the code was part of mouse handler at first).
* gui/settingsDialogs: remove extra right parenthesis from object coordinate control label.

josephsl committed:
Object coordinates/config, settings, event: address review actions.

Reviewed by @LeonardDer (Babbage): make object coordinate announcement a combo box:
* Config spec: object coordinates is now an option list.
* Settings: checkbox -> combo box, using same logic as progress bar output setting.
* NVDAObbjects/event_becomeNavigatorObject: detect different coordinate announcement settings and play coordinate beep if appropriate (for example, only play beeps if focus has moved and is told to play it when focus moves or both object navigation and focus).

josephsl committed:
Object coordinates: add an unassigned command in global commands to toggle this setting. Re #2559.

josephsl committed:
User guide: document the redesigned object coordinates combo box. Re #2559.

josephsl committed:
Mouse handler: remove blank line

josephsl committed:
Object coordinates/screen explorer: address review actions.

Reviewed by @LeonardDer 9Babbage): pull in multiple displays support from mouse handler. This reolsves a potential problem where coordinates will not be played if moving to an object located on secondary displays.

josephsl commented:
Hi,

To @LeonardDer: hope I addressed your comments. Thanks.

josephsl committed:
Merge branch 'master' into i2559

josephsl committed:
Object coordinates setting: log message typo fix

josephsl committed:
Merge branch 'master' into i2559

leonardder commented on source/globalCommands.py:
```diff
@@ -1463,6 +1463,29 @@ def script_toggleProgressBarOutput(self,gesture):
 	script_toggleProgressBarOutput.__doc__=_("Toggles between beeps, speech, beeps and speech, and off, for reporting progress bar updates")
 	script_toggleProgressBarOutput.category=SCRCAT_SPEECH
 
+	def script_togglePlayObjectCoordinates(self,gesture):
```
Could you please change this implementation to follow the implementation strategy of script_braille_toggleFocusContextPresentation? This includes creating constants for objFocus, objNav, both, etc.

Alternatively, you can choose the implementation of script_toggleReportLineIndentation.

feerrenrut replied:
I agree, perhaps define a dictionary first with the expected values and the ui messages, then you can get the index of the current `playObjectCoordinates` value increment it, and fetch the ui message from the dictionary for the new `playObjextCoordinates` value.

josephsl commented:
Hi, I see. I’d like to hear comments from @feerrenrut, as he did an overhaul of mouse handler before me. Thanks.

 

From: Leonard de Ruijter <notifications@github.com> 
Sent: Friday, July 20, 2018 10:36 AM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Author <author@noreply.github.com>
Subject: Re: [nvaccess/nvda] Object coordinates R2: rebased from master in 2018 Re #2559, #6078 (#8457)

 

@leonardder commented on this pull request.

  _____  

In source/screenExplorer.py <https://github.com/nvaccess/nvda/pull/8457#discussion_r204117452> :

>  import api
 import tones
 import controlTypes
 import textInfos
 import speech
+import screenBitmap
+import config
+import mouseHandler
+import wx
+
+def playLocationCoordinates(x, y, screenWidth, screenHeight, screenMinPos=None, detectBrightness=True,blurFactor=0):
+       """Plays a tone indicating the XY-coordinate for the given location. This works for keyboard-based object navigation/focus movement, mouse movement and during touch hover gesture.  
+       @param obj: the coordinate for the location and the screen resolution (typically desktopObject.location)
+       @type obj: int
+       """
+       # Have a fake wx.Point in case min pos is not defined on multiple monitors.
+       if screenMinPos is None:

To clarify, see the SM_CXVIRTUALSCREEN, SM_CYVIRTUALSCREEN, SM_XVIRTUALSCREEN and SM_YVIRTUALSCREEN. If I understand the code correctly (and probably I don't), throwing these four constants at GetSystemMetrics would make mouseHandler.getTotalWidthAndHeightAndMinimumPosition obsolete.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/8457#discussion_r204117452> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AHgLkELEprU7nd9nQ0GHOWks-gEqXW5kks5uIhTrgaJpZM4U59v5> .

josephsl committed:
Merge branch 'master' into i2559

josephsl commented:
Hi,

Regarding global commands: line indentation is done that way because the flags are stored across two keys instead of one. In a way, the implementation was borrowed from progress bar output toggle (NVDA+U), as this one does exactly the same thing.

Thanks.

josephsl committed:
Merge branch 'master' into i2559

josephsl committed:
Screen explorer/play coordinates: bring in documentation from old mouseHandler.playMouseCoordinates function so people can understand what the new function does. Re #2559.

josephsl committed:
Screen explorer/play object coordinates: use obj.location.center.

Thanks to location helper, it is possible to obtain object coordinates in a simple way through named tuples. One of these is obj.location.center, which is applicable to locating the center of the object for which coordinates should be played. Thus use this property from now on.

josephsl commented:
Hi,

obj.location.center is now used to obtain where the object's center lives, a nod to location helper integration.

Thanks.

feerrenrut commented on source/globalCommands.py:
```diff
@@ -1475,6 +1475,29 @@ def script_toggleProgressBarOutput(self,gesture):
+		if playObjectCoordinates == "both":
+			playObjectCoordinates = "off"
+			# Translators: A mode where object location coordinates will not be announced.
+			ui.message(_("Do not play Object location tone"))
```
The capital O is not needed on Object

feerrenrut commented on source/mouseHandler.py:
```diff
@@ -166,11 +135,8 @@ def executeMouseMoveEvent(x,y):
-		playAudioCoordinates(x, y, screenWidth, screenHeight, minPos,
-			config.conf['mouse']['audioCoordinates_detectBrightness'],
-			config.conf['mouse']['audioCoordinates_blurFactor'])
+		screenExplorer.playLocationCoordinates(x,y,screenWidth,screenHeight,minPos,config.conf['mouse']['audioCoordinates_detectBrightness'],config.conf['mouse']['audioCoordinates_blurFactor'])
```
This line is quite long, could you split up the args onto new lines.

feerrenrut commented on source/screenExplorer.py:
```diff
@@ -1,8 +1,69 @@
+	# Have a fake wx.Point in case min pos is not defined on multiple monitors.
+	if screenMinPos is None:
+		screenMinPos = wx.Point()
+	# Do not play any tone if offscreen.
```
The old version of this function had:

```
	# make relative to (0,0) and positive
	x = x - screenMinPos.x
	y = y - screenMinPos.y
```

Presumably this is still necessary? I believe that with multiple screens, negative x and y values are expected. `ScreenMinPos` is intended to allow these negative values to be normalized.

josephsl commented:
Hi, tone playback takes priority over speech and braille, and thus it might not be decoupled for a while. As for script to change this setting, it was borrowed from progress bar output toggle (NVDA+U). As for screen bitmap, I don’t know any add-ons using it (I’m wondering if Focus Highlight uses this), and if it turns out nobody is using it, then it should be removed the version after this PR is merged. As for coordinate limit calculation, I’ll think about it and see what can be done. Thanks.

 

From: Reef Turner <notifications@github.com> 
Sent: Monday, July 30, 2018 11:56 PM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Author <author@noreply.github.com>
Subject: Re: [nvaccess/nvda] Object coordinates R2: rebased from master in 2018 Re #2559, #6078 (#8457)

 

@feerrenrut commented on this pull request.

  _____  

In source/NVDAObjects/__init__.py <https://github.com/nvaccess/nvda/pull/8457#discussion_r206395026> :

> @@ -974,6 +975,11 @@ def event_becomeNavigatorObject(self, isFocus=False):
               @param isFocus: true if the navigator object was set due to a focus change.
               @type isFocus: bool
               """
+              # #2559: coordinate tone playback takes priority.

I don't really understand this comment. Takes priority over what?

The bigger question I have: Is this really the best place for this code? I doubt it's necessary to add a dependency on screenExplorer within the NVDAObjects package. Instead could this be decoupled?

  _____  

In source/globalCommands.py <https://github.com/nvaccess/nvda/pull/8457#discussion_r206405826> :

> @@ -1463,6 +1463,29 @@ def script_toggleProgressBarOutput(self,gesture):
        script_toggleProgressBarOutput.__doc__=_("Toggles between beeps, speech, beeps and speech, and off, for reporting progress bar updates")
        script_toggleProgressBarOutput.category=SCRCAT_SPEECH
 
+       def script_togglePlayObjectCoordinates(self,gesture):

I agree, perhaps define a dictionary first with the expected values and the ui messages, then you can get the index of the current playObjectCoordinates value increment it, and fetch the ui message from the dictionary for the new playObjextCoordinates value.

  _____  

In source/globalCommands.py <https://github.com/nvaccess/nvda/pull/8457#discussion_r206406032> :

> @@ -1475,6 +1475,29 @@ def script_toggleProgressBarOutput(self,gesture):
        script_toggleProgressBarOutput.__doc__=_("Toggles between beeps, speech, beeps and speech, and off, for reporting progress bar updates")
        script_toggleProgressBarOutput.category=SCRCAT_SPEECH
 
+       def script_togglePlayObjectCoordinates(self,gesture):
+              playObjectCoordinates = config.conf["presentation"]["playObjectCoordinates"]
+              if playObjectCoordinates == "both":
+                      playObjectCoordinates = "off"
+                      # Translators: A mode where object location coordinates will not be announced.
+                      ui.message(_("Do not play Object location tone"))

The capital O is not needed on Object

  _____  

In source/mouseHandler.py <https://github.com/nvaccess/nvda/pull/8457#discussion_r206407862> :

>       if config.conf["mouse"]["audioCoordinatesOnMouseMove"]:
-              playAudioCoordinates(x, y, screenWidth, screenHeight, minPos,
-                      config.conf['mouse']['audioCoordinates_detectBrightness'],
-                      config.conf['mouse']['audioCoordinates_blurFactor'])
+        screenExplorer.playLocationCoordinates(x,y,screenWidth,screenHeight,minPos,config.conf['mouse']['audioCoordinates_detectBrightness'],config.conf['mouse']['audioCoordinates_blurFactor'])

This line is quite long, could you split up the args onto new lines.

  _____  

In source/screenExplorer.py <https://github.com/nvaccess/nvda/pull/8457#discussion_r206408489> :

>  import api
 import tones
 import controlTypes
 import textInfos
 import speech
+import screenBitmap
+import config
+import mouseHandler
+import wx
+
+def playLocationCoordinates(x, y, screenWidth, screenHeight, screenMinPos=None, detectBrightness=True,blurFactor=0):
+       """Plays a tone indicating the XY-coordinate for the given location. This works for keyboard-based object navigation/focus movement, mouse movement and during touch hover gesture.  
+       @param obj: the coordinate for the location and the screen resolution (typically desktopObject.location)
+       @type obj: int
+       """
+       # Have a fake wx.Point in case min pos is not defined on multiple monitors.
+       if screenMinPos is None:
+              screenMinPos = wx.Point()
+       # Do not play any tone if offscreen.
+       if 0 <= x <= screenWidth or 0 <= y <= screenHeight:

Yeah, I would also prefer that the logic in this if statement was reversed and just had an early return. This reduces the indentation level of everything below.

  _____  

In source/screenExplorer.py <https://github.com/nvaccess/nvda/pull/8457#discussion_r206409887> :

> +import config
+import mouseHandler
+import wx
+
+def playLocationCoordinates(x, y, screenWidth, screenHeight, screenMinPos=None, detectBrightness=True,blurFactor=0):
+       """Plays a tone indicating the XY-coordinate for the given location.
+       This works for keyboard-based object navigation/focus movement, mouse movement and during touch hover gesture.
+       Left to right adjusting the volume between left and right speakers.
+       Top to bottom adjusts the pitch of the sound.
+       Brightness adjusts the volume of the sound (applicable for mouse movement).
+       Coordinates (x, y) are absolute, and can be negative.
+       """
+       # Have a fake wx.Point in case min pos is not defined on multiple monitors.
+       if screenMinPos is None:
+              screenMinPos = wx.Point()
+       # Do not play any tone if offscreen.

The old version of this function had:

        # make relative to (0,0) and positive
        x = x - screenMinPos.x
        y = y - screenMinPos.y

Presumably this is still necessary? I believe that with multiple screens, negative x and y values are expected. ScreenMinPos is intended to allow these negative values to be normalized.

  _____  

In source/screenExplorer.py <https://github.com/nvaccess/nvda/pull/8457#discussion_r206411858> :

> +     @type obj: int
+       """
+       # Have a fake wx.Point in case min pos is not defined on multiple monitors.
+       if screenMinPos is None:
+              screenMinPos = wx.Point()
+       # Do not play any tone if offscreen.
+       if 0 <= x <= screenWidth or 0 <= y <= screenHeight:
+              minPitch=config.conf['mouse']['audioCoordinates_minPitch']
+              maxPitch=config.conf['mouse']['audioCoordinates_maxPitch']
+              curPitch=minPitch+((maxPitch-minPitch)*((screenHeight-y)/float(screenHeight)))
+              if detectBrightness:
+                      startX=min(max(x-blurFactor,0),screenWidth)+screenMinPos.x
+                      startY=min(max(y-blurFactor,0),screenHeight)+screenMinPos.y
+                      width=min(blurFactor+1,screenWidth)
+                      height=min(blurFactor+1,screenHeight)
+                      grey=screenBitmap.rgbPixelBrightness(mouseHandler.scrBmpObj.captureImage( startX, startY, width, height)[0][0])

There are no other references to scrBmpObj outside of mouseHandler.py, you can confirm with git grep -i scrBmpObj.

Are you worried that addons are using it? My preference would be to understand its usage, and what the intention was behind initialising it this way, with the goal of moving it out of mouseHandler

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/8457#pullrequestreview-141800507> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AHgLkOUHBLVHznqN7vXtTTzr7wH6GAptks5uL_9xgaJpZM4U59v5> .

josephsl committed:
Merge branch 'master' into i2559

josephsl committed:
Global commands/object coordinates: Object -> object (lowercase).

Commented by Reef Turner (NV Access): lower case 'object'.

josephsl committed:
Screen explorer/play coordinates: simplify the whole thing by returning early if object is offscreen. Re #2559.

Reviewed by Reef Turner (NV Access): it is better to return early if things are offscreen. This also allows following lines to be less indented.

leonardder commented:
[**@josephsl**](https://github.com/josephsl) commented on [31 jul. 2018 09:05 CEST](https://github.com/nvaccess/nvda/pull/8457#issuecomment-409117709 "2018-07-31T07:05:00Z - Replied by Github Reply Comments"):
> As for script to change this setting, it was borrowed from progress bar output toggle (NVDA+U). 

My comment about this script still stands though. I'd still prefer a script that doesn't have al these if checks. I'd say you can leave the progress bar script itself alone.

> Then it should be removed the version after this PR is merged. 

@feerrenrut and I both seem to think that it makes sense to move it as part of this particular pr.

josephsl commented:
Hi,

After thinking about it (including code reuse, potential impact for translators, code consistency and what not), I'll adopt your recommendations:

* NVDA objects module will come with a dictionary that'll map config values to human-readable names.
* Both the global commands/script and Object Presentation/play object coordinates combo box will retrieve options via this new dictionary.
* I'll leave open the possibility of refactoring progress bar output script and progress bar output combo box via a follow-up pull request.
* I'll remove screen bitmap from mouse handler.

Thanks.

josephsl committed:
Merge branch 'master' into i2559

josephsl committed:
NVDA objects: add a map of settings messages for play object coordinates option. Re #2559.

Reviewed by Leonard de Ruijter (Babbage) and Reef Turner (NV Access): use a dedicated dictionary for storing object coordinate playback options. These settings will be found in NVDAObjects module.

josephsl committed:
NVDA objects/play coordinates map: clarify translator comments.

josephsl committed:
Global commands and Object Presentation settings panel: pull in play object ocordinates options from nVDAObjects module. Re #2559.

For global commands, play object coordinates toggle script has been rewritten to be based on the approach followed by braille focus context presentation script.
In Object Presentation settings panel, values and item labels for play object coordinates combo box comes from NVDAObjects module.

josephsl committed:
User guide: clarify play object coordinates combo box value labels. Re #2559.

josephsl committed:
What's new: add new features entry for play object coordinates setting.

josephsl commented:
Hi,

For both Reef and Leonard: I hope I addressed suggested changes.

Thanks.

josephsl commented:
Hi,

By the way, if this is pushed back to 2018.4, feel free to move the what's new entry for 2559 to 2018.4 section. Thanks for all your reviews.

feerrenrut commented on source/globalCommands.py:
```diff
@@ -1476,24 +1476,19 @@ def script_toggleProgressBarOutput(self,gesture):
+		# Translators: Reports the new state of object coordinate playback.
+		# %s will be replaced with play object coordinates setting.
+		# For example, the full message might be "Play object coordinates: when focus moves"
+		ui.message(_("Play object coordinates: %s")%labels[newIndex].lower())
```
I think this could be hard to translate in some languages. Why not have two sets of choices, one for here, and another for the settings dialog?

josephsl replied:
Hi,

Ah that... This is the same approach taken by braille focus context options, and is also seen for symbol level command (NVDA+P). As for translations, it is really up to translators to decide how these strings should be translated after consulting with the community.

Thanks.

feerrenrut commented on source/NVDAObjects/__init__.py:
```diff
@@ -974,6 +975,11 @@ def event_becomeNavigatorObject(self, isFocus=False):
 		@param isFocus: true if the navigator object was set due to a focus change.
 		@type isFocus: bool
 		"""
+		# #2559: coordinate tone playback takes priority.
```
Could you clarify the comment here?

josephsl committed:
What's new: remove play object coordinates entry.

josephsl committed:
Merge branch 'master' into i2559

josephsl committed:
Become navigator object event: clarify comment. Re #2559

josephsl committed:
Play object coordinates: use screen min pos attributes instead of using constants when comparing coordinates. Re #2559.

josephsl committed:
Merge branch 'master' into i2559

josephsl committed:
Merge branch 'master' into i2559

josephsl committed:
Merged master, fixing conflicts

josephsl committed:
Mouse audio coordinates: remove redundant function from mouse handler, as it is now taken care of by screen explorer. Re #2559

josephsl commented:
Hi,

I'm willing to yield this pull request to higher priority work. in the meantime, I'll generate a try build for adventurers to test with.

Thanks.

josephsl committed:
Merge branch 'master' into i2559

josephsl committed:
Merge branch 'master' into i2559

josephsl committed:
Merge branch 'master' into i2559

josephsl commented:
Hi,,
Six months later...

That high-priority work (Python 3) is done, so I'm coming back to this. I'll do something about this PR this week.

Thanks.

josephsl committed:
Merged master

josephsl committed:
Screen explorer/Python 3: xrange -> range. Re #2559

josephsl committed:
Object coordinates: lint work. Re #2559.

Lint work include:
* Global commands: convert the script to use script decorator.
* Spacing around operators.
* Split long lines.

josephsl committed:
Merged master

josephsl commented:
Hi,

Being considered for abandoning it in light of #11006. 

Thanks.

feerrenrut commented:
> Being considered for abandoning it in light of #11006.

If you are no longer interested in making this change, please do say so. Otherwise please continue to work on this PR. #11006 requests no new PR's. This PR already exists and is thus in the backlog of PR's that we want to address before accepting new ones. We are trying to stop PR's from going stale, and then becoming hard to merge. 

Forgive me for replying to you multiple times, I want to avoid the community being mislead.

josephsl commented:
Hi, I’ll do another rebase attempt and see if it is feasible to continue this work. Thanks.

 

From: Reef Turner <notifications@github.com> 
Sent: Wednesday, April 15, 2020 3:55 AM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Author <author@noreply.github.com>
Subject: Re: [nvaccess/nvda] Object coordinates R2: rebased from master in 2018 Re #2559, #6078 (#8457)

 

Being considered for abandoning it in light of #11006 <https://github.com/nvaccess/nvda/issues/11006> .

If you are no longer interested in making this change, please do say so. Otherwise please continue to work on this PR. #11006 <https://github.com/nvaccess/nvda/issues/11006>  requests no new PR's. This PR already exists and is thus in the backlog of PR's that we want to address before accepting new ones. We are trying to stop PR's from going stale, and then becoming hard to merge.

Forgive me for replying to you multiple times, I want to avoid the community being mislead.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/8457#issuecomment-613968258> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AB4AXEGBJUC5VJ2UNIT4ARLRMWG7XANCNFSM4FHH3P4Q> .

josephsl committed:
Merge branch 'master' into i2559

josephsl committed:
Merge branch 'master' into i2559

josephsl commented:
Hi,

Update: yep, the PR is back. At least until coordinates-based audio themes comes to NVDA (there is an add-on out there already), I think this PR would serve useful purposes.

Thanks.

josephsl committed:
Object coordinates script: move translator comment below description definition.

feerrenrut commented:
We have discussed this and have decided that although interesting it would be better as an addon to explore the user experience. If necessary, we would accept a PR to enable support for an addon such as this.

josephsl commented:
Hi, the add-on is already available: Object Location Tones, or better yet, Audio Themes 3D. Thanks.

 

From: Reef Turner <notifications@github.com> 
Sent: Thursday, July 2, 2020 11:57 PM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Author <author@noreply.github.com>
Subject: Re: [nvaccess/nvda] Object coordinates R2: rebased from master in 2018 Re #2559, #6078 (#8457)

 

We have discussed this and have decided that although interesting it would be better as an addon to explore the user experience. If necessary, we would accept a PR to enable support for an addon such as this.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/8457#issuecomment-653385852> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AB4AXEF2XPTKR5GDY7HNKSDRZV6KFANCNFSM4FHH3P4Q> .
